pkg/zcache/combined_cache.go

```diff
@@ -0,0 +1,56 @@
+package zcache
+
+import (
+	"context"
+	"time"
+)
+
+type CombinedCache interface {
+	ZCache
+}
+
+type combinedCache struct {
+	localCache         LocalCache
+	remoteCache        RemoteCache
+	isRemoteBestEffort bool
+}
+
+func (c *combinedCache) Set(ctx context.Context, key string, value interface{}, remoteCacheTtl time.Duration) error {
+	if err := c.remoteCache.Set(ctx, key, value, remoteCacheTtl); err != nil && !c.isRemoteBestEffort {
+		return err
+	}
+
+	// ttl is controlled by cache instantiation, so it does not matter here
+	if err := c.localCache.Set(ctx, key, value, -1); err != nil {
+		return err
```

Comment:
We should add metrics know if synchronization is lost between the cache, what do you think?